x-pack/plugins/apm/common/agent_configuration/runtime_types/storage_size_rt.test.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getStorageSizeRt } from './storage_size_rt';
+import { isRight } from 'fp-ts/lib/Either';
+import { PathReporter } from 'io-ts/lib/PathReporter';
+
+describe('storageSizeRt', () => {
+  describe('must accept any amount', () => {
+    const storageSizeRt = getStorageSizeRt({});
+    describe('it should not accept', () => {
+      ['MB', 1, '1', '5ZB', '6YB', '-2G'].map((input) => {
+        it(`${JSON.stringify(input)}`, () => {
+          expect(isRight(storageSizeRt.decode(input))).toBe(false);
+        });
+      });
+    });
+
+    describe('it should accept', () => {
+      ['1B', '0mb', '1b', '-2kb', '3mb', '10GB', '2TB'].map((input) => {
+        it(`${JSON.stringify(input)}`, () => {
+          expect(isRight(storageSizeRt.decode(input))).toBe(true);
+        });
+      });
+    });
+  });
+  describe('must be at least 2000kB', () => {
```

Comment:
nit: We shouldn't add "2000kB" to the spec title. This made me think that the implementation enforced a min size of 2000kB, when it's  not the case.

```suggestion
  describe('supports a min size', () => {
```